mantidproject/mantid
# HB3A Single point measurement peak integration

Implement a functionality in HB3A interface to calculate peak intensities of weak peaks which are only measured by single point.

**To test:**

Here is the workflow to calculate single-measurement peak intensity and export to Fullprof data file.

Using Exp 668 as a test case.

* A user specifies the range of scan numbers (289 - 450) that include all the neutron peaks regardless whether they are single-point measurement or multiple-points measurement.
* The user pushes button *Survey*.  Mantid will load all the SPICE files of the scans.
* The user sorts the table by sample temperature ('Sample Temp') to check that all the scans were measured under similar temperature.
* The user finds the multiple-measurements run with highest single-measurement counts and view the raw detector's counts in tab 'View Raw Data'.
* The user define and apply ROI from the strongest multi-measurement nuclear peak.

* The user integrate all multi-measurement scan peaks.  Mantid will generate a report for user to find out whether ROI is well defined or not. Mantid will also generate a report for counts outside of ROI for background.
* The user launches single-point peak integration window from menu *Peak Integraton* and sub menu *Single Pt Integration*. Mantid will add all the single-point scans to the popped out window.
* The user will be prompted with the information about how to map single-point scan to a complete nuclear peak scan with same 2theta.
* The user may load a CSV file to which integrated peak table was exported previously.
* The user pushes button *Retrieve FWHM*. Mantid then will gather FWHM value from integrated peak.
* The user add mapped scans for each single-pt scan to peak processint tab, and integrate peaks.
* The user pushes *Retrieve FHWHM* again to make sure every scan to have an FWHM.
* The user pushes *Integrate* to integrate peaks.
* The user pushes *Export to Table* to add the integrated peaks to peak processing table for final output.


* The user selects *All* single-point-measurement scans by selecting *Single Pt Scans* in combo-box



Fixes #21837. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
